vlog4j-graal/src/main/java/org/semanticweb/vlog4j/graal/GraalImporter.java

```diff
@@ -0,0 +1,214 @@
+	}
+
+	/**
+	 * Converts a {@link ConjunctiveQuery Graal Query} into a {@link ImportedGraalQuery}.
+	 * To use this with the {@link Reasoner}, add the {@code rule} from {@link ImportedGraalQuery#getRule()} as a Rule via {@link Reasoner#addRules(Rule...)}
+	 * and use it as the Atom for {@link Reasoner#answerQuery(Atom, boolean)}.
+	 * 
+	 * <p>
+	 * <b>WARNING</b>: The supplied {@code identifier} will be used to create a predicate containing all
+	 * answer variables from the {@code query}. If you use this identifier in another predicate, you will get conflicts.
+	 * </p>
+	 * @param identifier
+	 * @param query
+	 * @return
+	 */
+	public static ImportedGraalQuery importQuery(String identifier, ConjunctiveQuery query) {
+		Conjunction conjunction = importAtomSet(query.getAtomSet());
+		List<Term> answerVariables = importTerms(query.getAnswerVariables());
+		
+		return new  ImportedGraalQuery(identifier, answerVariables, conjunction);
+	}
+	
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Rule Graal Rule} into a {@link Rule Vlog4J Rule}.
+	 * @param rule A {@link fr.lirmm.graphik.graal.api.core.Rule Graal Rule}
+	 * @return A {@link Rule Vlog4J Rule}
+	 */
+	public static Rule importRule(fr.lirmm.graphik.graal.api.core.Rule rule) {
+		Conjunction head = importAtomSet(rule.getHead());
+		Conjunction body = importAtomSet(rule.getBody());
+		return makeRule(head, body);
+	}
+
+	/**
+	 * Converts a {@link List} of {@link fr.lirmm.graphik.graal.api.core.Rule Graal Rules} into a
+	 * {@link List} of {@link Rule VLog4J Rules}.
+	 * @param rules A {@link List} of {@link fr.lirmm.graphik.graal.api.core.Rule Graal Rules}.
+	 * @return A {@link List} of {@link Rule VLog4J Rules}.
+	 */
+	public static List<Rule> importRules(List<fr.lirmm.graphik.graal.api.core.Rule> rules) {
+		List<Rule> result = new ArrayList<>();
+		for (fr.lirmm.graphik.graal.api.core.Rule rule : rules) {
+			result.add(importRule(rule));;
+		}
+		return result;
+	}
+	
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Term Graal Term} into a {@link Term VLog4J Term}.
+	 * Tests if the term is a {@link fr.lirmm.graphik.graal.api.core.Term#isVariable() Variable} or {@link fr.lirmm.graphik.graal.api.core.Term#isConstant() Constant}
+	 * and converts accordingly. Throws a {@link GraalImportException} if it is neither.
+	 * @param term A {@link fr.lirmm.graphik.graal.api.core.Term Graal Term}
+	 * @return A {@link Term VLog4J Term}
+	 * @throws GraalImportException If the term is neither variable nor constant.
+	 */
+	private static Term importTerm(fr.lirmm.graphik.graal.api.core.Term term) {
+		if (term.isConstant())
+			return makeConstant(term.getIdentifier().toString());
+		else if (term.isVariable())
+			return makeVariable(term.getIdentifier().toString());
+		else
+			throw new GraalImportException(MessageFormat.format("Term {0} with identifier {1} could not be imported because it is neither constant nor variable.", term, term.getIdentifier().toString())); 
```

Comment:
It can be useful to also display the term label.

Reply:
Added the label to the error message.